src/Symfony/Bundle/SecurityBundle/Routing/LogoutRouteLoader.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\Routing;
+
+use Symfony\Component\DependencyInjection\Config\ContainerParametersResource;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+
+final class LogoutRouteLoader
+{
+    /**
+     * @param array<string, string> $logoutUris    Logout URIs indexed by the corresponding firewall name
+     * @param string                $parameterName Name of the container parameter containing {@see $logoutUris}' value
+     */
+    public function __construct(
+        private readonly array $logoutUris,
+        private readonly string $parameterName,
+    ) {
+    }
+
+    public function __invoke(): RouteCollection
+    {
+        $collection = new RouteCollection();
+        $collection->addResource(new ContainerParametersResource([$this->parameterName => $this->logoutUris]));
+
+        $routeNames = [];
+        foreach ($this->logoutUris as $firewallName => $logoutPath) {
+            $routeName = '_logout_'.$firewallName;
+
+            if (isset($routeNames[$logoutPath])) {
+                $collection->addAlias($routeNames[$logoutPath], $routeName);
+            } else {
+                $routeNames[$logoutPath] = $routeName;
+            }
+
+            $collection->add($routeName, new Route($logoutPath));
```

Comment:
I think we can do something here, to fix this issue : https://github.com/symfony/symfony/issues/51313

Reply:
Do you mean we should restrict these routes to the POST method?

Reply:
After reading @Spomky's comment https://github.com/symfony/symfony/issues/51313#issuecomment-1685219937, I would tend to agree with him

Reply:
There is more to this than just changing the routesʼ method. I don't think this PR is a good place to introduce such changes.

Reply:
You're right, it probably has an impact on the LogoutListener, LogoutUrlGenerator or LogoutUrlExtension.
It would be better to do a general PR on the topic